Sources/TensorFlow/Core/LazyTensorOperation.swift

```diff
@@ -374,9 +374,28 @@ extension LazyTensorOperation: TFTensorOperation {
         // If we want to stage this, we will need to add control dependencies.
         // For the time-being, just build a TFE_Op and run it.
         //
+        // Collect all the unmaterialized inputs.
+        var unmaterializedInputs = Array<LazyTensorOperation>()
+        unmaterializedInputs.reserveCapacity(inputs.count)
+        for input in inputs {
+            switch input {
+            case .single(let v):
+                if let lazyOperation = v.lazyTensorOperation {
+                    unmaterializedInputs.append(lazyOperation)
+                }
+            case .list(let values):
+                for value in values {
```

Comment:
Consider using [`Array.compactMap(_:)`](https://developer.apple.com/documentation/swift/array/2957701-compactmap):
```swift
                unmaterializedInputs.append(
                    contentsOf: values.compactMap { $0.lazyTensorOperation }
                )
```

Reply:
Thanks for the suggestion, @dan-zheng However, isn't the explicit loop slightly more efficient as we don't need to create the temporary array?

Reply:
> Thanks for the suggestion, @dan-zheng. However, isn't the explicit loop slightly more efficient as we don't need to create the temporary array?

Yes, oops. You can use a [`LazySequence`](https://developer.apple.com/documentation/swift/lazysequence), which will not materialize a temporary array:
```diff
                 unmaterializedInputs.append(
-                    contentsOf: values.compactMap { $0.lazyTensorOperation }
+                    contentsOf: values.lazy.compactMap { $0.lazyTensorOperation }
                 )
```

Reply:
Gotcha, thanks!